src/Firebase/AppCheck/AppCheckTokenGenerator.php

```diff
@@ -0,0 +1,53 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Kreait\Firebase\AppCheck;
+
+use Beste\Clock\SystemClock;
+use Firebase\JWT\JWT;
+use Kreait\Firebase\Exception\AppCheck\InvalidAppCheckTokenOptions;
+use Kreait\Firebase\ServiceAccount;
+use StellaMaris\Clock\ClockInterface;
+
+/**
+ * @internal
+ */
+class AppCheckTokenGenerator
+{
+    private const APP_CHECK_AUDIENCE = 'https://firebaseappcheck.googleapis.com/google.firebase.appcheck.v1.TokenExchangeService';
+    private ServiceAccount $serviceAccount;
+    private ClockInterface $clock;
+
+    public function __construct(ServiceAccount $serviceAccount, ?ClockInterface $clock = null)
+    {
+        $this->serviceAccount = $serviceAccount;
+        $this->clock = $clock ?? SystemClock::create();
+    }
+
+    /**
+     * @param non-empty-string $appId the Application Id to use for the generated token
+     *
+     * @throws InvalidAppCheckTokenOptions
+     *
+     * @return string the generated token
+     */
+    public function createCustomToken(string $appId, ?AppCheckTokenOptions $options = null): string
+    {
+        $now = $this->clock->now()->getTimestamp();
+        $payload = [
+            'iss' => $this->serviceAccount->getClientEmail(),
+            'sub' => $this->serviceAccount->getClientEmail(),
+            'app_id' => $appId,
+            'aud' => self::APP_CHECK_AUDIENCE,
+            'iat' => $now,
+            'exp' => $now + 300,
```

Comment:
I based this off of what the official node package is doing. They are always setting the expiration for the custom token to 300. 

I think because it creates a custom token and then immediately exchanges the custom token for a App Check token, that the time here doesn't matter much. I'm guessing they chose 5 mins because it's long enough to ensure a race condition doesn't occur here with the token exchange but it's short enough that it is a short lived token?

Reply:
Sounds reasonable, thanks for clarifying! 🙌